test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java

```diff
@@ -0,0 +1,272 @@
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUndo_ContainsSimpleCommands() 
+    {
+        aCompoundCommand1.execute();
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(5, 5, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+        try 
+        {
+            ArrayList<Node> aListNodesToBeRemoved = (ArrayList<Node>) (aNodesToBeRemoved.get(aGraph));
+            assertFalse(aListNodesToBeRemoved.contains((Node) aClassNode));
+            ArrayList<Edge> aListEdgesToBeRemoved = (ArrayList<Edge>) aEdgesToBeRemoved.get(aGraph);
+            assertFalse(aListEdgesToBeRemoved.contains(aCallEdge));
+        } 
+        catch (IllegalArgumentException | IllegalAccessException e1) 
+        {
+            fail();
+        }
+        
+        aCompoundCommand1.undo();
+        
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(0, 0, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+        try 
+        {
+            ArrayList<Node> aListNodesToBeRemoved = (ArrayList<Node>) (aNodesToBeRemoved.get(aGraph));
+            assertTrue(aListNodesToBeRemoved.contains((Node) aClassNode));
+            ArrayList<Edge> aListEdgesToBeRemoved = (ArrayList<Edge>) aEdgesToBeRemoved.get(aGraph);
+            assertTrue(aListEdgesToBeRemoved.contains(aCallEdge));
+        } 
+        catch (IllegalArgumentException | IllegalAccessException e1) 
+        {
+            fail();
+        }
+    }
+    
+    @Test
+    public void testExecute_ContainsCompoundCommands() 
+    {
+    	assertFalse(aGraph.getRootNodes().contains(aClassNode));
+        assertFalse(aGraph.getEdges().contains(aCallEdge));
+        assertFalse(aGraph.getRootNodes().contains(aActorNode));
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(0, 0, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+        assertEquals(aActorNode.getBounds(), new Rectangle2D.Double(0, 0, aActorNode.getBounds().getWidth(), aActorNode.getBounds().getHeight()));
+        
+        aCompoundCommand3.execute();        
+        
+        assertTrue(aGraph.getRootNodes().contains(aClassNode));
+        assertTrue(aGraph.getEdges().contains(aCallEdge));
+        assertTrue(aGraph.getRootNodes().contains(aActorNode));
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(2, 8, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+        assertEquals(aActorNode.getBounds(), new Rectangle2D.Double(0, 0, aActorNode.getBounds().getWidth(), aActorNode.getBounds().getHeight()));
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUndo_ContainsCompoundCommands() 
+    {
+        aCompoundCommand3.execute();
+        
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(2, 8, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
```

Comment:
same comment, you can remove the baseline predicates (everything until the SUT execution, i.e., the call to `undo`